esp-config/src/generate.rs

```diff
@@ -149,10 +149,17 @@ fn emit_configuration(
     // emit cfgs and set envs
     for (name, value) in configs.into_iter() {
         let cfg_name = snake_case(name.trim_start_matches(prefix));
-        println!("cargo:rustc-check-cfg=cfg({cfg_name})");
         match value {
-            Value::Bool(true) => {
-                println!("cargo:rustc-cfg={cfg_name}")
+            Value::Bool(b) => {
+                println!("cargo:rustc-check-cfg=cfg({cfg_name})");
+                if *b {
+                    println!("cargo:rustc-cfg={cfg_name}")
+                }
+            }
+            Value::String(v) => {
+                // TODO we need to emit the other option else we'll get a lint warning...
+                println!("cargo:rustc-check-cfg=cfg({cfg_name}, values(\"{v}\"))");
```

Comment:
This might be a pain to support in esp-config, so I might still remove this in favour of making this runtime configurable option that we pass into `esp_wifi::init` - thoughts?

Reply:
Had to look up that this is needed for the powersave config. IMO it should be configurable, and if we make that happen, an esp-config key won't make much sense just to select the default value.

Reply:
I agree, I'll rework it into a runtime config.